modin/pandas/__init__.py

```diff
@@ -21,8 +21,8 @@
 
     if pandas.__version__ != __pandas_version__:
         warnings.warn(
-            f"The pandas version installed {pandas.__version__} does not match the supported pandas version in"
-            + f" Modin {__pandas_version__} compatibility mode. This may cause undesired side effects!"
```

Comment:
Maybe we shouldn't remove the mention of `compatibility mode`?

Reply:
Yeah I agree.

Reply:
Good point. Should we call it "Python {version} legacy compatibility mode" or similar instead? I feel like "compatibility mode" alone is a little confusing.

Reply:
@noloerino your suggestion sounds good

Reply:
I've made the change.